pkg/runtime/reconciler.go

```diff
@@ -268,49 +268,113 @@ func (r *resourceReconciler) Sync(
+			return err
+		}
+		for _, condition := range latest.Conditions() {
+			if condition.Type == ackv1alpha1.ConditionTypeResourceSynced {
+				if condition.Status == corev1.ConditionTrue {
+					if duration := r.rmf.RequeueOnSuccessSeconds(); duration > 0 {
+						rlog.Debug(
+							"requeueing resource after resource synced condition true",
+						)
+						return requeue.NeededAfter(nil, time.Duration(duration)*time.Second)
+					}
+				} else {
 					rlog.Debug(
-						"requeueing resource after resource synced condition true",
+						"requeueing resource after finding resource synced condition false",
 					)
-					return requeue.NeededAfter(nil, time.Duration(duration)*time.Second)
+					return requeue.NeededAfter(
+						ackerr.TemporaryOutOfSync, requeue.DefaultRequeueAfterDuration)
 				}
-			} else {
-				rlog.Debug(
-					"requeueing resource after finding resource synced condition false",
-				)
-				return requeue.NeededAfter(
-					ackerr.TemporaryOutOfSync, requeue.DefaultRequeueAfterDuration)				
 			}
 		}
 	}
 	return nil
 }
 
 // patchResource patches the custom resource in the Kubernetes API to match the
-// supplied latest resource.
+// supplied latest resource's "metadata", "spec" and "status"
+// To only patch the metadata and spec, use "patchResourceMetadataAndSpec".
+// To only patch the status, use "patchResourceStatus"
 func (r *resourceReconciler) patchResource(
 	ctx context.Context,
 	desired acktypes.AWSResource,
 	latest acktypes.AWSResource,
+) error {
+	err := r.patchResourceMetadataAndSpec(ctx, desired, latest)
+	if err != nil {
+		return err
+	}
+
+	err = r.patchResourceStatus(ctx, desired, latest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// patchResourceMetadataAndSpec patches the custom resource in the Kubernetes API to match the
+// supplied latest resource's metadata and spec.
+func (r *resourceReconciler) patchResourceMetadataAndSpec(
+	ctx context.Context,
+	desired acktypes.AWSResource,
+	latest acktypes.AWSResource,
 ) error {
 	var err error
 	rlog := ackrtlog.FromContext(ctx)
-	exit := rlog.Trace("r.patchResource")
+	exit := rlog.Trace("r.patchResourceMetadataAndSpec")
+	defer exit(err)
+
+	desiredObjAnnotations := make(map[string]string)
+	latestObjAnnotations := make(map[string]string)
+
+	if desired.MetaObject().GetAnnotations() != nil {
+		desiredObjAnnotations = desired.MetaObject().GetAnnotations()
+	}
+
+	if latest.MetaObject().GetAnnotations() != nil {
+		latestObjAnnotations = latest.MetaObject().GetAnnotations()
+	}
+
+	if !ackcompare.MapStringStringEqual(desiredObjAnnotations, latestObjAnnotations) || r.rd.Delta(desired, latest).DifferentAt("Spec") {
+		rlog.Enter("kc.Patch (metadata + spec)")
+		err = r.kc.Patch(
+			ctx,
+			latest.RuntimeObject(),
+			client.MergeFrom(desired.RuntimeObject()),
+		)
+		rlog.Exit("kc.Patch (metadata + spec)", err)
+		if err != nil {
+			return err
+		}
+		rlog.Debug("patched resource metadata and spec", "latest", latest)
+	} else {
```

Comment:
nit: in L350 you could check for `ackcompare.MapStringStringEqual(desiredObjAnnotations, latestObjAnnotations) && !r.rd.Delta(desired, latest).DifferentAt("Spec")` and exit early - to avoid using the `else` statement.